ibm/service/contextbasedrestrictions/resource_ibm_cbr_rule.go

```diff
@@ -362,99 +308,109 @@ func resourceIBMCbrRuleCreate(context context.Context, d *schema.ResourceData, m
 
 	d.SetId(*rule.ID)
 
-	// handle Eventual consistency case
-	readNewCbrRule(contextBasedRestrictionsClient, context, d)
-
-	return resourceIBMCbrRuleRead(context, d, meta)
-}
-
-func resourceIBMCbrRuleRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	contextBasedRestrictionsClient, err := meta.(conns.ClientSession).ContextBasedRestrictionsV1()
-	if err != nil {
-		return diag.FromErr(err)
+	if err := resourceIBMCbrRuleSetData(response, rule, d); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting rule's resource data: %s", err))
 	}
 
-	getRuleOptions := &contextbasedrestrictionsv1.GetRuleOptions{}
-
-	getRuleOptions.SetRuleID(d.Id())
+	return nil
+}
 
-	rule, response, err := contextBasedRestrictionsClient.GetRuleWithContext(context, getRuleOptions)
-	if err != nil {
-		if response != nil && response.StatusCode == 404 {
-			d.SetId("")
-			return nil
-		}
-		log.Printf("[DEBUG] GetRuleWithContext failed %s\n%s", err, response)
-		return diag.FromErr(fmt.Errorf("GetRuleWithContext failed %s\n%s", err, response))
-	}
-
-	if err = d.Set("x_correlation_id", getRuleOptions.XCorrelationID); err != nil {
-		return diag.FromErr(fmt.Errorf("Error setting x_correlation_id: %s", err))
-	}
-	if err = d.Set("transaction_id", getRuleOptions.TransactionID); err != nil {
-		return diag.FromErr(fmt.Errorf("Error setting transaction_id: %s", err))
-	}
-	if err = d.Set("description", rule.Description); err != nil {
-		return diag.FromErr(fmt.Errorf("Error setting description: %s", err))
+func resourceIBMCbrRuleSetData(response *core.DetailedResponse, rule *contextbasedrestrictionsv1.Rule, d *schema.ResourceData) error {
```

Comment:
It would be better to have the parameter order for `resourceIBMCbrRuleSetData` be consistent with the return value order of the methods that provide the values (e.g. `CreateRuleWithContext`, `ReplaceRuleWithContext`, `GetRuleWithContext`)

```suggestion
func resourceIBMCbrRuleSetData(rule *contextbasedrestrictionsv1.Rule, response *core.DetailedResponse, d *schema.ResourceData) error {
```

Reply:
updated to the suggestion